File: send_recv.py
from mpi4py import MPI

comm = MPI.COMM_WORLD
rank = comm.Get_rank()
size = comm.Get_size()

if size < 2:
    print("World size must be greater than 1 for this example")
    MPI.COMM_WORLD.Abort(1)
    


if rank == 0:
    message = "Hello Process 1!"
    # Send the message to process 1
    comm.send(message, dest=1, tag=0)
    print("Process 0 sending message to process 1")
elif rank == 1:
    # Receive the message from process 0
    received_message = comm.recv(source=0, tag=0)
    print(f"Process 1 received message: {received_message}")
else:
    print(f"Process {rank} is not involved in this communication.")

File: send_recv.c
#include <mpi.h>
#include <stdio.h>

// int MPI_Send(const void *buf, int count, MPI_Datatype datatype, int dest, int tag, MPI_Comm comm)

// int MPI_Recv(void *buf, int count, MPI_Datatype datatype, int source, int tag,
//              MPI_Comm comm, MPI_Status *status)

int main(int argc, char **argv)
{
    int world_rank;
    int world_size;
    int message;
    int received_message;
    MPI_Status status;

    MPI_Init(NULL, NULL);

    MPI_Comm_rank(MPI_COMM_WORLD, &world_rank);

    MPI_Comm_size(MPI_COMM_WORLD, &world_size)

        if (world_size < 2)
    {
        fprintf(stderr, "World size must be greater than 1 for this example\n");
        MPI_Abort(MPI_COMM_WORLD, 1);
    }

    if (world_rank == 0)
    {
        message = 123;
        MPI_Send(&message, 1, MPI_INT, 1, 0, MPI_COMM_WORLD);
        printf("Process 0 sending message to process 1\n");
    }
    else if (world_rank == 1)
    {
        // Receive the message from process 0
        MPI_Recv(&received_message, 1, MPI_INT, 0, 0, MPI_COMM_WORLD, &status);
        printf("Process 1 received message: %d\n", received_message);
    }
    else
    {
        printf("Process %d is not involved in this communication.\n", world_rank);
    }

    MPI_Finalize();
}